pkg/monitor/monitor/template.go

```diff
@@ -304,137 +258,207 @@ func scrapeJob(jobName string, componentPattern config.Regexp, cmodel *MonitorCo
+		clusterTargetPattern := cluster.Name
+
+		nsTargetPattern := cluster.Namespace
+
+		schemeRelabelConfig := yaml.MapItem{
+			Key:   "scheme",
+			Value: "http",
+		}
+		tlsConfigRelabelConfig := yaml.MapSlice{
+			{
+				Key:   "insecure_skip_verify",
+				Value: true,
+			},
+		}
+
+		if cluster.enableTLS && !isDMJob(jobName) {
+			schemeRelabelConfig.Value = "https"
+			// lightning does not need to authenticate the access of other components,
+			// so there is no need to enable mtls for the time being.
+			if jobName != "lightning" {
+				tcTlsSecretName := util.ClusterClientTLSSecretName(cluster.Name)
+				tlsConfigRelabelConfig = yaml.MapSlice{
+					yaml.MapItem{
+						Key:   "ca_file",
+						Value: path.Join(util.ClusterAssetsTLSPath, TLSAssetKey{"secret", cluster.Namespace, tcTlsSecretName, corev1.ServiceAccountRootCAKey}.String()),
+					},
+					yaml.MapItem{
+						Key:   "cert_file",
+						Value: path.Join(util.ClusterAssetsTLSPath, TLSAssetKey{"secret", cluster.Namespace, tcTlsSecretName, corev1.TLSCertKey}.String()),
+					},
+					yaml.MapItem{
+						Key:   "key_file",
+						Value: path.Join(util.ClusterAssetsTLSPath, TLSAssetKey{"secret", cluster.Namespace, tcTlsSecretName, corev1.TLSPrivateKeyKey}.String()),
+					},
+				}
+			}
 		}
-		nsTargetPattern, err := config.NewRegexp(cluster.Namespace)
-		if err != nil {
-			klog.Errorf("generate scrapeJob[%s] clusterName:%s namespace:%s error:%v", jobName, cluster.Name, cluster.Namespace, err)
-			continue
+
+		if cluster.enableTLS && isDMJob(jobName) {
+			schemeRelabelConfig.Value = "https"
+			dmTlsSecretName := util.DMClientTLSSecretName(cluster.Name)
+			tlsConfigRelabelConfig = yaml.MapSlice{
+				yaml.MapItem{
+					Key:   "ca_file",
+					Value: path.Join(util.ClusterAssetsTLSPath, TLSAssetKey{"secret", cluster.Namespace, dmTlsSecretName, corev1.ServiceAccountRootCAKey}.String()),
+				},
+				yaml.MapItem{
+					Key:   "cert_file",
+					Value: path.Join(util.ClusterAssetsTLSPath, TLSAssetKey{"secret", cluster.Namespace, dmTlsSecretName, corev1.TLSCertKey}.String()),
+				},
+				yaml.MapItem{
+					Key:   "key_file",
+					Value: path.Join(util.ClusterAssetsTLSPath, TLSAssetKey{"secret", cluster.Namespace, dmTlsSecretName, corev1.TLSPrivateKeyKey}.String()),
+				},
+			}
+
 		}
 
-		scrapeconfig := &config.ScrapeConfig{
-			JobName:        fmt.Sprintf("%s-%s-%s", cluster.Namespace, cluster.Name, jobName),
-			ScrapeInterval: model.Duration(15 * time.Second),
-			Scheme:         "http",
-			HonorLabels:    true,
-			ServiceDiscoveryConfig: config.ServiceDiscoveryConfig{
-				KubernetesSDConfigs: []*config.KubernetesSDConfig{
+		scrapeconfig := yaml.MapSlice{
+			{Key: "job_name", Value: fmt.Sprintf("%s-%s-%s", cluster.Namespace, cluster.Name, jobName)},
+			{Key: "honor_labels", Value: true},
+			{Key: "scrape_interval", Value: "15s"},
+			schemeRelabelConfig,
+
+			{Key: "kubernetes_sd_configs", Value: []yaml.MapSlice{
+				{
+					{
+						Key:   "api_server",
```

Comment:
no need to add this item?

Reply:
Before it have `api_server` configuration, ref : https://github.com/pingcap/tidb-operator/blob/bedbbfa536fb1280d782aec26b09115a524c3499/pkg/monitor/monitor/template_test.go#L95